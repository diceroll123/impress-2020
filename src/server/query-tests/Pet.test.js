const gql = require("graphql-tag");
const {
  query,
  getDbCalls,
  clearDbCalls,
  useTestDb,
  connectToDb,
} = require("./setup.js");

describe("Pet", () => {
  it("looks up a pet", async () => {
    const res = await query({
      query: gql`
        query {
          petOnNeopetsDotCom(petName: "roopal27") {
            species {
              id
              name
            }
            color {
              id
              name
            }
            pose
            items {
              id
              name
              description
              thumbnailUrl
              rarityIndex
              isNc
            }
          }
        }
      `,
    });

    expect(res).toHaveNoErrors();
    expect(res.data).toMatchSnapshot();
    expect(getDbCalls()).toMatchInlineSnapshot(`
      Array [
        Array [
          "SELECT * FROM species_translations
             WHERE species_id IN (?) AND locale = \\"en\\"",
          Array [
            54,
          ],
        ],
        Array [
          "SELECT * FROM color_translations
             WHERE color_id IN (?) AND locale = \\"en\\"",
          Array [
            75,
          ],
        ],
      ]
    `);
  });

  it("models new pet and item data", async () => {
    useTestDb();

    const res = await query({
      query: gql`
        query {
          petOnNeopetsDotCom(petName: "roopal27") {
            items {
              id
              name
              description
              thumbnailUrl
              rarityIndex
              isNc
            }
          }
        }
      `,
    });

    expect(res).toHaveNoErrors();
    expect(res.data).toMatchSnapshot();
    expect(getDbCalls()).toMatchSnapshot();

    clearDbCalls();

    const res2 = await query({
      query: gql`
        query {
          petAppearance(colorId: "75", speciesId: "54", pose: SAD_MASC) {
            id
            pose
            layers {
              id
              swfUrl
            }
            restrictedZones {
              id
            }
          }

          items(
            ids: [
              "37229"
              "37375"
              "38911"
              "38912"
              "38913"
              "43014"
              "43397"
              "48313"
            ]
          ) {
            id
            name
            description
            thumbnailUrl
            rarityIndex
            isNc
            createdAt
          }
        }
      `,
    });

    expect(res2).toHaveNoErrors();
    expect(res2.data).toMatchSnapshot();
    expect(getDbCalls()).toMatchSnapshot();
  });

  it("models updated item data", async () => {
    useTestDb();

    // First, write a fake version of the Jewelled Staff to the database.
    // It's mostly the real data, except we changed rarity_index,
    // thumbnail_url, translated name, and translated description.
    const db = await connectToDb();
    await Promise.all([
      db.query(
        `INSERT INTO items (id, zones_restrict, thumbnail_url, category,
                            type, rarity_index, price, weight_lbs)
           VALUES (43397, "00000000000000000000000000000000000000000000000",
                   "http://example.com/favicon.ico", "Clothes", "Clothes", 101,
                   0, 1);`
      ),
      db.query(
        `INSERT INTO item_translations (item_id, locale, name, description,
                                        rarity)
           VALUES (43397, "en", "Bejewelled Staffo",
                   "This staff is really neat and good!", "Artifact")`
      ),
    ]);

    clearDbCalls();

    // Then, load a pet wearing this. It should trigger an UPDATE for the item
    // and its translation, and return the new names in the query.
    const res = await query({
      query: gql`
        query {
          petOnNeopetsDotCom(petName: "roopal27") {
            items {
              id
              name
              description
              thumbnailUrl
              rarityIndex
            }
          }
        }
      `,
    });

    expect(res).toHaveNoErrors();
    const itemData = res.data.petOnNeopetsDotCom.items.find(
      (item) => item.id === "43397"
    );
    expect(itemData).toEqual({
      id: "43397",
      name: "Jewelled Staff",
      description: "This jewelled staff shines with a magical light.",
      thumbnailUrl: "http://images.neopets.com/items/mall_staff_jewelled.gif",
      rarityIndex: 500,
    });
    expect(getDbCalls()).toMatchSnapshot();

    clearDbCalls();

    // Finally, load the item. It should have the updated values.
    const res2 = await query({
      query: gql`
        query {
          item(id: "43397") {
            id
            name
            description
            thumbnailUrl
            rarityIndex
          }
        }
      `,
    });

    expect(res2).toHaveNoErrors();
    expect(res2.data.item).toEqual({
      id: "43397",
      name: "Jewelled Staff",
      description: "This jewelled staff shines with a magical light.",
      thumbnailUrl: "http://images.neopets.com/items/mall_staff_jewelled.gif",
      rarityIndex: 500,
    });
    expect(getDbCalls()).toMatchSnapshot();
  });
});
