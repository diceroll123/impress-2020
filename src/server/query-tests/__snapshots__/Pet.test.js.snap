// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pet looks up a pet 1`] = `
Object {
  "petOnNeopetsDotCom": Object {
    "color": Object {
      "id": "75",
      "name": "Starry",
    },
    "items": Array [
      Object {
        "description": "What does this ball actually do?",
        "id": "37229",
        "isNc": false,
        "name": "Magic Ball Table",
        "rarityIndex": 101,
        "thumbnailUrl": "http://images.neopets.com/items/gif_magicball_table.gif",
      },
      Object {
        "description": "Dont forget to wish upon a star.",
        "id": "37375",
        "isNc": false,
        "name": "Moon and Stars Background",
        "rarityIndex": 75,
        "thumbnailUrl": "http://images.neopets.com/items/bg_moonstars.gif",
      },
      Object {
        "description": "Hide your face and hair so no one can recognise you.",
        "id": "38911",
        "isNc": false,
        "name": "Zafara Agent Hood",
        "rarityIndex": 92,
        "thumbnailUrl": "http://images.neopets.com/items/clo_zafara_agent_hood.gif",
      },
      Object {
        "description": "This robe is great for being stealthy.",
        "id": "38912",
        "isNc": false,
        "name": "Zafara Agent Robe",
        "rarityIndex": 90,
        "thumbnailUrl": "http://images.neopets.com/items/clo_zafara_agent_robe.gif",
      },
      Object {
        "description": "Dont leave any trace that you were there with these gloves.",
        "id": "38913",
        "isNc": false,
        "name": "Zafara Agent Gloves",
        "rarityIndex": 88,
        "thumbnailUrl": "http://images.neopets.com/items/clo_zafara_agent_gloves.gif",
      },
      Object {
        "description": "These leaves almost look magical with their gentle glow.",
        "id": "43014",
        "isNc": false,
        "name": "Green Leaf String Lights",
        "rarityIndex": 80,
        "thumbnailUrl": "http://images.neopets.com/items/toy_stringlight_illleaf.gif",
      },
      Object {
        "description": "This jewelled staff shines with a magical light.",
        "id": "43397",
        "isNc": true,
        "name": "Jewelled Staff",
        "rarityIndex": 500,
        "thumbnailUrl": "http://images.neopets.com/items/mall_staff_jewelled.gif",
      },
      Object {
        "description": "Even the announcers of the Altador Cup celebrate.  This was given out by the Advent Calendar in Y11.",
        "id": "48313",
        "isNc": false,
        "name": "Altador Cup Brooch",
        "rarityIndex": 101,
        "thumbnailUrl": "http://images.neopets.com/items/clo_altcuplogo_brooch.gif",
      },
    ],
    "pose": "SAD_MASC",
    "species": Object {
      "id": "54",
      "name": "Zafara",
    },
  },
}
`;

exports[`Pet models new pet and item data 1`] = `
Object {
  "petOnNeopetsDotCom": Object {
    "items": Array [
      Object {
        "description": "What does this ball actually do?",
        "id": "37229",
        "isNc": false,
        "name": "Magic Ball Table",
        "rarityIndex": 101,
        "thumbnailUrl": "http://images.neopets.com/items/gif_magicball_table.gif",
      },
      Object {
        "description": "Dont forget to wish upon a star.",
        "id": "37375",
        "isNc": false,
        "name": "Moon and Stars Background",
        "rarityIndex": 75,
        "thumbnailUrl": "http://images.neopets.com/items/bg_moonstars.gif",
      },
      Object {
        "description": "Hide your face and hair so no one can recognise you.",
        "id": "38911",
        "isNc": false,
        "name": "Zafara Agent Hood",
        "rarityIndex": 92,
        "thumbnailUrl": "http://images.neopets.com/items/clo_zafara_agent_hood.gif",
      },
      Object {
        "description": "This robe is great for being stealthy.",
        "id": "38912",
        "isNc": false,
        "name": "Zafara Agent Robe",
        "rarityIndex": 90,
        "thumbnailUrl": "http://images.neopets.com/items/clo_zafara_agent_robe.gif",
      },
      Object {
        "description": "Dont leave any trace that you were there with these gloves.",
        "id": "38913",
        "isNc": false,
        "name": "Zafara Agent Gloves",
        "rarityIndex": 88,
        "thumbnailUrl": "http://images.neopets.com/items/clo_zafara_agent_gloves.gif",
      },
      Object {
        "description": "These leaves almost look magical with their gentle glow.",
        "id": "43014",
        "isNc": false,
        "name": "Green Leaf String Lights",
        "rarityIndex": 80,
        "thumbnailUrl": "http://images.neopets.com/items/toy_stringlight_illleaf.gif",
      },
      Object {
        "description": "This jewelled staff shines with a magical light.",
        "id": "43397",
        "isNc": true,
        "name": "Jewelled Staff",
        "rarityIndex": 500,
        "thumbnailUrl": "http://images.neopets.com/items/mall_staff_jewelled.gif",
      },
      Object {
        "description": "Even the announcers of the Altador Cup celebrate.  This was given out by the Advent Calendar in Y11.",
        "id": "48313",
        "isNc": false,
        "name": "Altador Cup Brooch",
        "rarityIndex": 101,
        "thumbnailUrl": "http://images.neopets.com/items/clo_altcuplogo_brooch.gif",
      },
    ],
  },
}
`;

exports[`Pet models new pet and item data 2`] = `
Array [
  Array [
    "SELECT * FROM pet_types WHERE (species_id = ? AND color_id = ?)",
    Array [
      "54",
      "75",
    ],
  ],
  Array [
    "SELECT * FROM items WHERE id IN (?,?,?,?,?,?,?,?)",
    Array [
      "37229",
      "37375",
      "38911",
      "38912",
      "38913",
      "43014",
      "43397",
      "48313",
    ],
  ],
  Array [
    "SELECT * FROM item_translations WHERE item_id IN (?,?,?,?,?,?,?,?) AND locale = \\"en\\"",
    Array [
      "37229",
      "37375",
      "38911",
      "38912",
      "38913",
      "43014",
      "43397",
      "48313",
    ],
  ],
  Array [
    "SELECT * FROM swf_assets WHERE (type = ? AND remote_id = ?) OR (type = ? AND remote_id = ?) OR (type = ? AND remote_id = ?) OR (type = ? AND remote_id = ?) OR (type = ? AND remote_id = ?) OR (type = ? AND remote_id = ?) OR (type = ? AND remote_id = ?) OR (type = ? AND remote_id = ?)",
    Array [
      "object",
      "6829",
      "object",
      "14855",
      "object",
      "14856",
      "object",
      "14857",
      "object",
      "36414",
      "object",
      "39646",
      "object",
      "51959",
      "object",
      "56478",
    ],
  ],
  Array [
    "INSERT INTO pet_types (body_id, color_id, created_at, species_id) VALUES (?, ?, ?, ?);",
    Array [
      "180",
      "75",
      2020-01-01T00:00:00.000Z,
      "54",
    ],
  ],
  Array [
    "INSERT INTO items (category, created_at, id, price, rarity_index, thumbnail_url, type, updated_at, weight_lbs, zones_restrict) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?, ?, ?, ?);",
    Array [
      "Gift",
      2020-01-01T00:00:00.000Z,
      "37229",
      0,
      101,
      "http://images.neopets.com/items/gif_magicball_table.gif",
      "Gift",
      2020-01-01T00:00:00.000Z,
      1,
      "0000000000000000000000000000000000000000000000",
      "Special",
      2020-01-01T00:00:00.000Z,
      "37375",
      209,
      75,
      "http://images.neopets.com/items/bg_moonstars.gif",
      "Mystical Surroundings",
      2020-01-01T00:00:00.000Z,
      1,
      "0000000000000000000000000000000000000000000000000000",
      "Clothes",
      2020-01-01T00:00:00.000Z,
      "38911",
      980,
      92,
      "http://images.neopets.com/items/clo_zafara_agent_hood.gif",
      "Clothes",
      2020-01-01T00:00:00.000Z,
      1,
      "0000000000000000000000000000000000001100000000000000",
      "Clothes",
      2020-01-01T00:00:00.000Z,
      "38912",
      1476,
      90,
      "http://images.neopets.com/items/clo_zafara_agent_robe.gif",
      "Clothes",
      2020-01-01T00:00:00.000Z,
      1,
      "0000000000000000000101000000000000000000000000000000",
      "Clothes",
      2020-01-01T00:00:00.000Z,
      "38913",
      1177,
      88,
      "http://images.neopets.com/items/clo_zafara_agent_gloves.gif",
      "Clothes",
      2020-01-01T00:00:00.000Z,
      1,
      "0000000000000000000000000000000000000000000000000000",
      "Toys",
      2020-01-01T00:00:00.000Z,
      "43014",
      1033,
      80,
      "http://images.neopets.com/items/toy_stringlight_illleaf.gif",
      "Toy",
      2020-01-01T00:00:00.000Z,
      1,
      "0000000000000000000000000000000000000000000000",
      "Clothes",
      2020-01-01T00:00:00.000Z,
      "43397",
      0,
      500,
      "http://images.neopets.com/items/mall_staff_jewelled.gif",
      "Clothes",
      2020-01-01T00:00:00.000Z,
      1,
      "0000000000000000000000000000000000000000000000",
      "Clothes",
      2020-01-01T00:00:00.000Z,
      "48313",
      0,
      101,
      "http://images.neopets.com/items/clo_altcuplogo_brooch.gif",
      "Clothes",
      2020-01-01T00:00:00.000Z,
      1,
      "0000000000000000000000000000000000000000000000000000",
    ],
  ],
  Array [
    "INSERT INTO item_translations (created_at, description, item_id, locale, name, rarity, updated_at) VALUES (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?);",
    Array [
      2020-01-01T00:00:00.000Z,
      "What does this ball actually do?",
      "37229",
      "en",
      "Magic Ball Table",
      "Special",
      2020-01-01T00:00:00.000Z,
      2020-01-01T00:00:00.000Z,
      "Dont forget to wish upon a star.",
      "37375",
      "en",
      "Moon and Stars Background",
      "Uncommon",
      2020-01-01T00:00:00.000Z,
      2020-01-01T00:00:00.000Z,
      "Hide your face and hair so no one can recognise you.",
      "38911",
      "en",
      "Zafara Agent Hood",
      "Very Rare",
      2020-01-01T00:00:00.000Z,
      2020-01-01T00:00:00.000Z,
      "This robe is great for being stealthy.",
      "38912",
      "en",
      "Zafara Agent Robe",
      "Very Rare",
      2020-01-01T00:00:00.000Z,
      2020-01-01T00:00:00.000Z,
      "Dont leave any trace that you were there with these gloves.",
      "38913",
      "en",
      "Zafara Agent Gloves",
      "Rare",
      2020-01-01T00:00:00.000Z,
      2020-01-01T00:00:00.000Z,
      "These leaves almost look magical with their gentle glow.",
      "43014",
      "en",
      "Green Leaf String Lights",
      "Uncommon",
      2020-01-01T00:00:00.000Z,
      2020-01-01T00:00:00.000Z,
      "This jewelled staff shines with a magical light.",
      "43397",
      "en",
      "Jewelled Staff",
      "Artifact",
      2020-01-01T00:00:00.000Z,
      2020-01-01T00:00:00.000Z,
      "Even the announcers of the Altador Cup celebrate.  This was given out by the Advent Calendar in Y11.",
      "48313",
      "en",
      "Altador Cup Brooch",
      "Special",
      2020-01-01T00:00:00.000Z,
    ],
  ],
  Array [
    "INSERT INTO swf_assets (body_id, created_at, remote_id, type, url, zone_id, zones_restrict) VALUES (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?);",
    Array [
      180,
      2020-01-01T00:00:00.000Z,
      "6829",
      "object",
      "http://images.neopets.com/cp/items/swf/000/000/006/6829_1707e50385.swf",
      3,
      "",
      180,
      2020-01-01T00:00:00.000Z,
      "14855",
      "object",
      "http://images.neopets.com/cp/items/swf/000/000/014/14855_215f367070.swf",
      25,
      "",
      180,
      2020-01-01T00:00:00.000Z,
      "14856",
      "object",
      "http://images.neopets.com/cp/items/swf/000/000/014/14856_46c1b32797.swf",
      26,
      "",
      180,
      2020-01-01T00:00:00.000Z,
      "14857",
      "object",
      "http://images.neopets.com/cp/items/swf/000/000/014/14857_d43380ef66.swf",
      40,
      "",
      180,
      2020-01-01T00:00:00.000Z,
      "36414",
      "object",
      "http://images.neopets.com/cp/items/swf/000/000/036/36414_1e2aaab4ad.swf",
      48,
      "",
      180,
      2020-01-01T00:00:00.000Z,
      "39646",
      "object",
      "http://images.neopets.com/cp/items/swf/000/000/039/39646_e129e22ada.swf",
      42,
      "",
      180,
      2020-01-01T00:00:00.000Z,
      "51959",
      "object",
      "http://images.neopets.com/cp/items/swf/000/000/051/51959_4439727c48.swf",
      45,
      "",
      180,
      2020-01-01T00:00:00.000Z,
      "56478",
      "object",
      "http://images.neopets.com/cp/items/swf/000/000/056/56478_eabc28e7c7.swf",
      27,
      "",
    ],
  ],
]
`;

exports[`Pet models new pet and item data 3`] = `
Object {
  "items": Array [
    Object {
      "createdAt": "2020-01-01T00:00:00.000Z",
      "description": "What does this ball actually do?",
      "id": "37229",
      "isNc": false,
      "name": "Magic Ball Table",
      "rarityIndex": 101,
      "thumbnailUrl": "http://images.neopets.com/items/gif_magicball_table.gif",
    },
    Object {
      "createdAt": "2020-01-01T00:00:00.000Z",
      "description": "Dont forget to wish upon a star.",
      "id": "37375",
      "isNc": false,
      "name": "Moon and Stars Background",
      "rarityIndex": 75,
      "thumbnailUrl": "http://images.neopets.com/items/bg_moonstars.gif",
    },
    Object {
      "createdAt": "2020-01-01T00:00:00.000Z",
      "description": "Hide your face and hair so no one can recognise you.",
      "id": "38911",
      "isNc": false,
      "name": "Zafara Agent Hood",
      "rarityIndex": 92,
      "thumbnailUrl": "http://images.neopets.com/items/clo_zafara_agent_hood.gif",
    },
    Object {
      "createdAt": "2020-01-01T00:00:00.000Z",
      "description": "This robe is great for being stealthy.",
      "id": "38912",
      "isNc": false,
      "name": "Zafara Agent Robe",
      "rarityIndex": 90,
      "thumbnailUrl": "http://images.neopets.com/items/clo_zafara_agent_robe.gif",
    },
    Object {
      "createdAt": "2020-01-01T00:00:00.000Z",
      "description": "Dont leave any trace that you were there with these gloves.",
      "id": "38913",
      "isNc": false,
      "name": "Zafara Agent Gloves",
      "rarityIndex": 88,
      "thumbnailUrl": "http://images.neopets.com/items/clo_zafara_agent_gloves.gif",
    },
    Object {
      "createdAt": "2020-01-01T00:00:00.000Z",
      "description": "These leaves almost look magical with their gentle glow.",
      "id": "43014",
      "isNc": false,
      "name": "Green Leaf String Lights",
      "rarityIndex": 80,
      "thumbnailUrl": "http://images.neopets.com/items/toy_stringlight_illleaf.gif",
    },
    Object {
      "createdAt": "2020-01-01T00:00:00.000Z",
      "description": "This jewelled staff shines with a magical light.",
      "id": "43397",
      "isNc": true,
      "name": "Jewelled Staff",
      "rarityIndex": 500,
      "thumbnailUrl": "http://images.neopets.com/items/mall_staff_jewelled.gif",
    },
    Object {
      "createdAt": "2020-01-01T00:00:00.000Z",
      "description": "Even the announcers of the Altador Cup celebrate.  This was given out by the Advent Calendar in Y11.",
      "id": "48313",
      "isNc": false,
      "name": "Altador Cup Brooch",
      "rarityIndex": 101,
      "thumbnailUrl": "http://images.neopets.com/items/clo_altcuplogo_brooch.gif",
    },
  ],
  "petAppearance": null,
}
`;

exports[`Pet models new pet and item data 4`] = `
Array [
  Array [
    "SELECT * FROM pet_types WHERE (species_id = ? AND color_id = ?)",
    Array [
      "54",
      "75",
    ],
  ],
  Array [
    "SELECT * FROM item_translations WHERE item_id IN (?,?,?,?,?,?,?,?) AND locale = \\"en\\"",
    Array [
      "37229",
      "37375",
      "38911",
      "38912",
      "38913",
      "43014",
      "43397",
      "48313",
    ],
  ],
  Array [
    "SELECT * FROM items WHERE id IN (?,?,?,?,?,?,?,?)",
    Array [
      "37229",
      "37375",
      "38911",
      "38912",
      "38913",
      "43014",
      "43397",
      "48313",
    ],
  ],
]
`;

exports[`Pet models updated item data 1`] = `
Array [
  Array [
    "SELECT * FROM pet_types WHERE (species_id = ? AND color_id = ?)",
    Array [
      "54",
      "75",
    ],
  ],
  Array [
    "SELECT * FROM items WHERE id IN (?,?,?,?,?,?,?,?)",
    Array [
      "37229",
      "37375",
      "38911",
      "38912",
      "38913",
      "43014",
      "43397",
      "48313",
    ],
  ],
  Array [
    "SELECT * FROM item_translations WHERE item_id IN (?,?,?,?,?,?,?,?) AND locale = \\"en\\"",
    Array [
      "37229",
      "37375",
      "38911",
      "38912",
      "38913",
      "43014",
      "43397",
      "48313",
    ],
  ],
  Array [
    "SELECT * FROM swf_assets WHERE (type = ? AND remote_id = ?) OR (type = ? AND remote_id = ?) OR (type = ? AND remote_id = ?) OR (type = ? AND remote_id = ?) OR (type = ? AND remote_id = ?) OR (type = ? AND remote_id = ?) OR (type = ? AND remote_id = ?) OR (type = ? AND remote_id = ?)",
    Array [
      "object",
      "6829",
      "object",
      "14855",
      "object",
      "14856",
      "object",
      "14857",
      "object",
      "36414",
      "object",
      "39646",
      "object",
      "51959",
      "object",
      "56478",
    ],
  ],
  Array [
    "INSERT INTO pet_types (body_id, color_id, created_at, species_id) VALUES (?, ?, ?, ?);",
    Array [
      "180",
      "75",
      2020-01-01T00:00:00.000Z,
      "54",
    ],
  ],
  Array [
    "INSERT INTO items (category, created_at, id, price, rarity_index, thumbnail_url, type, updated_at, weight_lbs, zones_restrict) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?, ?, ?, ?);",
    Array [
      "Gift",
      2020-01-01T00:00:00.000Z,
      "37229",
      0,
      101,
      "http://images.neopets.com/items/gif_magicball_table.gif",
      "Gift",
      2020-01-01T00:00:00.000Z,
      1,
      "0000000000000000000000000000000000000000000000",
      "Special",
      2020-01-01T00:00:00.000Z,
      "37375",
      209,
      75,
      "http://images.neopets.com/items/bg_moonstars.gif",
      "Mystical Surroundings",
      2020-01-01T00:00:00.000Z,
      1,
      "0000000000000000000000000000000000000000000000000000",
      "Clothes",
      2020-01-01T00:00:00.000Z,
      "38911",
      980,
      92,
      "http://images.neopets.com/items/clo_zafara_agent_hood.gif",
      "Clothes",
      2020-01-01T00:00:00.000Z,
      1,
      "0000000000000000000000000000000000001100000000000000",
      "Clothes",
      2020-01-01T00:00:00.000Z,
      "38912",
      1476,
      90,
      "http://images.neopets.com/items/clo_zafara_agent_robe.gif",
      "Clothes",
      2020-01-01T00:00:00.000Z,
      1,
      "0000000000000000000101000000000000000000000000000000",
      "Clothes",
      2020-01-01T00:00:00.000Z,
      "38913",
      1177,
      88,
      "http://images.neopets.com/items/clo_zafara_agent_gloves.gif",
      "Clothes",
      2020-01-01T00:00:00.000Z,
      1,
      "0000000000000000000000000000000000000000000000000000",
      "Toys",
      2020-01-01T00:00:00.000Z,
      "43014",
      1033,
      80,
      "http://images.neopets.com/items/toy_stringlight_illleaf.gif",
      "Toy",
      2020-01-01T00:00:00.000Z,
      1,
      "0000000000000000000000000000000000000000000000",
      "Clothes",
      2020-01-01T00:00:00.000Z,
      "48313",
      0,
      101,
      "http://images.neopets.com/items/clo_altcuplogo_brooch.gif",
      "Clothes",
      2020-01-01T00:00:00.000Z,
      1,
      "0000000000000000000000000000000000000000000000000000",
    ],
  ],
  Array [
    "INSERT INTO item_translations (created_at, description, item_id, locale, name, rarity, updated_at) VALUES (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?);",
    Array [
      2020-01-01T00:00:00.000Z,
      "What does this ball actually do?",
      "37229",
      "en",
      "Magic Ball Table",
      "Special",
      2020-01-01T00:00:00.000Z,
      2020-01-01T00:00:00.000Z,
      "Dont forget to wish upon a star.",
      "37375",
      "en",
      "Moon and Stars Background",
      "Uncommon",
      2020-01-01T00:00:00.000Z,
      2020-01-01T00:00:00.000Z,
      "Hide your face and hair so no one can recognise you.",
      "38911",
      "en",
      "Zafara Agent Hood",
      "Very Rare",
      2020-01-01T00:00:00.000Z,
      2020-01-01T00:00:00.000Z,
      "This robe is great for being stealthy.",
      "38912",
      "en",
      "Zafara Agent Robe",
      "Very Rare",
      2020-01-01T00:00:00.000Z,
      2020-01-01T00:00:00.000Z,
      "Dont leave any trace that you were there with these gloves.",
      "38913",
      "en",
      "Zafara Agent Gloves",
      "Rare",
      2020-01-01T00:00:00.000Z,
      2020-01-01T00:00:00.000Z,
      "These leaves almost look magical with their gentle glow.",
      "43014",
      "en",
      "Green Leaf String Lights",
      "Uncommon",
      2020-01-01T00:00:00.000Z,
      2020-01-01T00:00:00.000Z,
      "Even the announcers of the Altador Cup celebrate.  This was given out by the Advent Calendar in Y11.",
      "48313",
      "en",
      "Altador Cup Brooch",
      "Special",
      2020-01-01T00:00:00.000Z,
    ],
  ],
  Array [
    "INSERT INTO swf_assets (body_id, created_at, remote_id, type, url, zone_id, zones_restrict) VALUES (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?), (?, ?, ?, ?, ?, ?, ?);",
    Array [
      180,
      2020-01-01T00:00:00.000Z,
      "6829",
      "object",
      "http://images.neopets.com/cp/items/swf/000/000/006/6829_1707e50385.swf",
      3,
      "",
      180,
      2020-01-01T00:00:00.000Z,
      "14855",
      "object",
      "http://images.neopets.com/cp/items/swf/000/000/014/14855_215f367070.swf",
      25,
      "",
      180,
      2020-01-01T00:00:00.000Z,
      "14856",
      "object",
      "http://images.neopets.com/cp/items/swf/000/000/014/14856_46c1b32797.swf",
      26,
      "",
      180,
      2020-01-01T00:00:00.000Z,
      "14857",
      "object",
      "http://images.neopets.com/cp/items/swf/000/000/014/14857_d43380ef66.swf",
      40,
      "",
      180,
      2020-01-01T00:00:00.000Z,
      "36414",
      "object",
      "http://images.neopets.com/cp/items/swf/000/000/036/36414_1e2aaab4ad.swf",
      48,
      "",
      180,
      2020-01-01T00:00:00.000Z,
      "39646",
      "object",
      "http://images.neopets.com/cp/items/swf/000/000/039/39646_e129e22ada.swf",
      42,
      "",
      180,
      2020-01-01T00:00:00.000Z,
      "51959",
      "object",
      "http://images.neopets.com/cp/items/swf/000/000/051/51959_4439727c48.swf",
      45,
      "",
      180,
      2020-01-01T00:00:00.000Z,
      "56478",
      "object",
      "http://images.neopets.com/cp/items/swf/000/000/056/56478_eabc28e7c7.swf",
      27,
      "",
    ],
  ],
  Array [
    "UPDATE items SET rarity_index = ?, thumbnail_url = ?, updated_at = ?, zones_restrict = ? WHERE id = ?;",
    Array [
      500,
      "http://images.neopets.com/items/mall_staff_jewelled.gif",
      2020-01-01T00:00:00.000Z,
      "0000000000000000000000000000000000000000000000",
      "43397",
    ],
  ],
  Array [
    "UPDATE item_translations SET description = ?, name = ?, updated_at = ? WHERE item_id = ? AND locale = \\"en\\";",
    Array [
      "This jewelled staff shines with a magical light.",
      "Jewelled Staff",
      2020-01-01T00:00:00.000Z,
      "43397",
    ],
  ],
]
`;

exports[`Pet models updated item data 2`] = `
Array [
  Array [
    "SELECT * FROM item_translations WHERE item_id IN (?) AND locale = \\"en\\"",
    Array [
      "43397",
    ],
  ],
  Array [
    "SELECT * FROM items WHERE id IN (?)",
    Array [
      "43397",
    ],
  ],
]
`;
